arch.go

```diff
@@ -194,25 +190,20 @@ func (e *archEngine) FilterList(archIds []archetypeId, comp []componentId) []arc
 	// TODO: could I maybe do something more optimal with archetypeMask?
 	// New way: With archSets that are just slices
 	// Logic: Go thorugh and keep track of how many times we see each archetype. Then only keep the archetypes that we've seen an amount of times equal to the number of components. If we have 5 components and see 5 for a specific archId, it means that each component has that archId
-	// TODO: this may be more efficient to use a slice?
-
-	// Clearing Optimization: https://go.dev/doc/go1.11#performance-compiler
-	for k := range e.archCount {
-		delete(e.archCount, k)
-	}
 
+	archCount := make([]int, len(e.lookup))
```

Comment:
I am somewhat concerned about the allocation here. But I understand why it's needed for multithreading at a system level. The allocation is actually not a big deal for desktop games, but because the game I'm developing runs in a browser (which is single threaded), every garbage collection operation blocks the game from running. On desktop this isn't a big deal because the mark phase (IIRC) runs in parallel with the game simulation, but in browser the mark phase ends up blocking the simulation from progressing which can cause dropped frames. I think the solution here is probably more broadly that all of the allocations should be in a world-specific memory arena, but I haven't read up on how those work yet.

Reply:
yeah. maybe not to allocate with `make` on the heap if lookup is relativelly small